### DISCUSSION TITLE
Should HTML documents produced by Quarto be tidy-HTML5 compliant?

### DISCUSSION THREAD

**Discussion by IndrajeetPatil** [quarto-dev/quarto-cli#3831] [Feature Requests]
Tidy HTML (http://www.html-tidy.org/) tool helps correct and clean up HTML and XML documents by fixing markup errors, and I am wondering if it might be a good idea to make sure that all HTML outputs from Quarto have `0 warnings` and `0 errors` from `tidy` check and also the formatting is `tidy`-compliant.

E.g. with [v1.3.78](https://github.com/quarto-dev/quarto-cli/releases/tag/v1.3.78), if I open example html format document shipped with RStudio and render it. The HTML produced has `<meta>` tags outside `<head>` tags:

```html
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.78">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

...
```

If I run tidy tool on this document:

```sh
tidy -m -i -w 120 -c -utf8 *.html
```

It changes this to:

```html
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="quarto-1.3.78" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />

...
```

But maybe the original HTML produced Quarto should already look tidied? 
That is, the users shouldn't even need to use tidy tool since Quarto HTMLs will by default be tidy-compliant, or at least as compliant as they can be without having to sacrifice any Quarto-specific innovation.

**Comment by dragonstyle** [comment]
It looks to me like there is a head tag in the first example (at the end of the line). Are there other issues that tidy flags?

**Reply by IndrajeetPatil** [reply]
Ah, missed that tag completely because of the formatting! 

There are a number of other changes, but most of them have to do with formatting.

Apart from formatting, with the default HTML document example rendered, `tidy` generates the following two warnings:

```sh
line 134 column 333 - Warning: trimming empty <i>
line 134 column 274 - Warning: trimming empty <button>
Info: Document content looks like XHTML5
Tidy found 2 warnings and 0 errors!
```

**Reply by dragonstyle** [reply]
I'll take a peek at those warnings, but I'm guessing that those are elements targeted by CSS or JS which need to remain (e.g. a bootstrap icon for example).

**Comment by maelle** [comment]
Related: https://github.com/quarto-dev/quarto-cli/discussions/7489

**Reply by cscheid** [reply]
I'd be cautious of trusting third party validators that have opinions about good HTML rather than actually being a validator.

For example, we document that if we fail to parse an HTML table that passes the [official w3c validator](https://validator.w3.org/), then [we consider that a bug](https://quarto.org/docs/prerelease/1.3/tables.html#limitations). But we can't take third-party tools as arbitrators of good markup, it's just not a good idea.

**Reply by maelle** [reply]
Thank you for your insight! :pray:
